src/layout/layout.rs

```diff
@@ -2236,6 +2283,32 @@ mod tests {
             assert_eq!(expected, r);
         }
 
+        #[rstest]
+        #[case::length_overlap(vec![(0  , 20) , (20 , 20) , (40 , 20)] , vec![Length(20) , Length(20) , Length(20)] , Flex::Start        , 0)]
+        #[case::length_overlap(vec![(0  , 20) , (19 , 20) , (38 , 20)] , vec![Length(20) , Length(20) , Length(20)] , Flex::Start        , 1)]
+        #[case::length_overlap(vec![(21 , 20) , (40 , 20) , (59 , 20)] , vec![Length(20) , Length(20) , Length(20)] , Flex::Center       , 1)]
+        #[case::length_overlap(vec![(42 , 20) , (61 , 20) , (80 , 20)] , vec![Length(20) , Length(20) , Length(20)] , Flex::End          , 1)]
+        #[case::length_overlap(vec![(0  , 20) , (19 , 20) , (38 , 62)] , vec![Length(20) , Length(20) , Length(20)] , Flex::Legacy       , 1)]
+        #[case::length_overlap(vec![(0  , 34) , (33 , 34) , (66 , 34)] , vec![Length(20) , Length(20) , Length(20)] , Flex::SpaceBetween , 1)]
+        #[case::length_overlap(vec![(0  , 34) , (33 , 34) , (66 , 34)] , vec![Length(20) , Length(20) , Length(20)] , Flex::SpaceAround  , 1)]
+        fn flex_overlap(
+            #[case] expected: Vec<(u16, u16)>,
+            #[case] constraints: Vec<Constraint>,
+            #[case] flex: Flex,
+            #[case] overlap: u16,
+        ) {
+            let rect = Rect::new(0, 0, 100, 1);
+            let r = Layout::horizontal(constraints)
+                .flex(flex)
+                .overlap(overlap.into())
+                .split(rect);
+            let result = r
+                .iter()
+                .map(|r| (r.x, r.width))
```

Comment:
Not related to this specific test, but the output of these tests when they fail is particularly poor. Would be nice to have a small struct for this that has start and width instead of the tuple.

E.g.
```
thread 'layout::layout::tests::split::fill_overlap::case_07_flex10' panicked at src/layout/layout.rs:2488:13:
assertion failed: `(left == right)`

Diff < left / right > :
 [
     (
         0,
<        55,
>        50,
     ),
     (
<        45,
<        55,
>        50,
>        50,
     ),
 ]
```

Reply:
I agree, the tests failing here are hard to debug. What about something like this?

<img width="840" alt="image" src="https://github.com/user-attachments/assets/59127bb9-8984-4fc8-9417-b43bc4b19fb2">

https://github.com/ratatui/ratatui/pull/1398/commits/03be41a02bee5ff0be386b4f4003aa94e60f4bbc

Reply:
Perhaps check starts and widths separately? It's tough to know what to do here (and it's a bit of a micro-optimization I guess)

I can see what you're doing there. The idea is good - perhaps could work by showing the rects as full rects with borders.

Reply:
I think regardless of what we do here that should happen in a separate PR, because we have to change every test here to use the new format.

Reply:
I've opened an issue to track this: https://github.com/ratatui/ratatui/issues/1400